### PR TITLE
Add tests for partial wal file generation

### PR DESCRIPTION
**Pull request by kaknikhil**
As part of https://github.com/greenplum-db/gpdb/pull/15187, we added partial wal file tests to gpdb6. Since partial wal file is already supported by gpdb7, we just need to port the tap tests from the aforementioned PR.

Partial wal scenario:

A few previous commits de7688442f and 7cbee7c introduced the concept of creating and archiving partial wal files when the mirror is promoted while the primary is still alive.  In this case, the last wal file on the old timeline in the mirror's pg_wal dir is renamed with the suffix ".partial".  This partial file also get archived. The original wal file only gets archived once the user runs pg_rewind.

Consider the following example: Let's assume that 0000100004 is the current wal file on the master

start with a master and standby pair
add data to master
contents of pg_wal on master:
      0000100001
      .....
      0000100003
      0000100004 - current wal file on master

master is alive but standby gets promoted
contents of pg_wal on standby:
      0000100001
      ....
      0000100003
      0000100004.partial (note that 0000100004 does not exist on the standby)
      0000200004 - current wal file on standby

Contents of the archive location:
      0000100003.tar.gz
      0000100004.partial.tar.gz

stop master with pg_ctl stop -m fast
contents of pg_wal on master:
      0000100004 on the master gets flushed and gets archived
      0000100004.done gets created on master
Contents of the archive location:
      0000100003.tar.gz
      0000100004.partial.tar.gz
      0000100004.tar.gz
run pg_rewind
      copies from standby to master
      removes 0000100004 and 0000100004.done from master's pg_wal dir

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/recovery/t/002_archiving.pl [hunk 1]
```diff
@@ -52,45 +52,209 @@
...
-my $node_standby2_data = $node_standby2->data_dir;
-
-like(
-	$log_contents,
-	qr/restored log file "00000002.history" from archive/s,
-	"00000002.history retrieved from the archives");
-ok( !-f "$node_standby2_data/pg_wal/RECOVERYHISTORY",
-	"RECOVERYHISTORY removed after promotion");
-ok( !-f "$node_standby2_data/pg_wal/RECOVERYXLOG",
-	"RECOVERYXLOG removed after promotion");
```
Comment (huansong) [inline src/test/recovery/t/002_archiving.pl:96]: Wondering why is this test removed (might be a copy/paste mistake)?

Comment (kaknikhil) [reply src/test/recovery/t/002_archiving.pl:96]: The tests weren’t removed, just moved into a new function named `check_history_files`

Comment (huansong) [reply src/test/recovery/t/002_archiving.pl:96]: I see, saw this comment block not there so thought the test got lost:
```
# Wait until the history file has been stored on the archives of the
# primary once the promotion of the standby completes.  This ensures that
# the second standby created below will be able to restore this file,
# creating a RECOVERYHISTORY.
```

**Review by huansong — APPROVED** [review]
LGTM!

File: src/test/recovery/t/002_archiving.pl (hunk 1 above)
Comment (kaknikhil) [reply src/test/recovery/t/002_archiving.pl:96]: Good point, I'll add this comment

**Pull request merged**